java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/aggregation/PreAggFuncWithLimit.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 The Feathub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf.aggregation;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+
+import java.util.LinkedList;
+
+/** The function that performs the pre-aggregation process for aggregation functions with limit. */
+public class PreAggFuncWithLimit<T>
```

Comment:
Can you add unit tests for `PreAggFuncWithLimit` and `AggFuncWithLmit`? 

I think we are missing an e2e test case for sliding window with limit. Can you also add an e2e test case to avoid regression in the future?